### PR TITLE
Add auto-merge capability to dlang-bot

### PR DESCRIPTION
**Pull request by wilzbach**

### CODE REVIEW

File: data/payloads/github_repos_dlang_dmd_issues?state=open&labels=auto-merge,auto-merge-squash [hunk 1]
```diff
@@ -0,0 +1,67 @@
+[
```
Comment (wilzbach) [inline data/payloads/github_repos_dlang_dmd_issues?state=open&labels=auto-merge,auto-merge-squash]: Sorry - with the issue API we can directly get the correct subset :)

File: source/app.d [hunk 2]
```diff
@@ -38,6 +42,8 @@ void startServer(HTTPServerSettings settings)
     settings.bindAddresses = ["0.0.0.0"];
     settings.options = HTTPServerOption.defaults & ~HTTPServerOption.parseJsonBody;
 
+    lastFullPRCheck = SysTime(Date(0, 1, 1));
```
Comment (wilzbach) [inline source/app.d]: FYI: `SysTime.min` is there, but didn't work. I guess it underflows?

File: test/utils.d [hunk 3]
```diff
@@ -65,14 +65,17 @@ auto payloadServer(scope HTTPServerRequest req, scope HTTPServerResponse res)
     auto idx = apiExpectations.map!(x => x.url).countUntil(req.requestURL);
     if (idx >= 0)
     {
-        expectation = apiExpectations[0];
+        expectation = apiExpectations[idx];
```
Comment (wilzbach) [inline test/utils.d]: Funnily I only found this when testing with this set of tests :P

File: source/app.d [hunk 4]
```diff
@@ -248,6 +297,121 @@ void updateGithubComment(string action, IssueRef[] refs, Issue[] descs, string c
...
+            labelUrl ~= "auto-merge";
+        if (labelInfo.hasAutoMergeSquash)
+            labelUrl ~= "auto-merge-squash";
+        ghSendRequest(HTTPMethod.DELETE, labelUrl);
+    }
+}
+
+void searchForAutoMergePrs(string repoSlug)
+{
+    auto issues = ghGetRequest("%s/repos/%s/issues?state=open&labels=auto-merge,auto-merge-squash"
```
Comment (MartinNowak) [inline source/app.d]: Unfortunately this is a logical AND between the labels, see https://api.github.com/repos/dlang/phobos/issues?state=open&labels=blocked,dependency. Searched the net, there doesn't seem to be an OR, so we'd need 2 queries (and maybe an intersection :/).

File: source/app.d (hunk 2 above)
Comment (MartinNowak) [inline source/app.d]: Supposed to work, and Duration can represent any differences.
https://dlang.org/library/std/datetime/sys_time.min.html
```d
import std.datetime;
Clock.currTime - SysTime.min;
```
-1419840 weeks, -3 days, -2 hours, -21 minutes, -37 secs, -762 ms, -353 μs, and -8 hnsecs

Comment (MartinNowak) [inline source/app.d]: You can use MonoTime instead.

Comment (MartinNowak) [inline source/app.d]: Or check relatively `lastFullPRCheck + 5.minutes > Clock.currTime`.

Comment (MartinNowak) [inline source/app.d]: Not important though

Comment (MartinNowak) [inline source/app.d]: SysTime.init is zero and won't overflow, both SysTime and Duration use long hnsecs internally.

File: source/app.d (hunk 4 above)
Comment (wilzbach) [inline source/app.d]: > Unfortunately this is a logical AND between the labels, 

Wow good catch! -> updated

**Comment by MartinNowak** [comment]
Will deploy this later.

**Pull request merged**